cdk/types/src/lib.rs

```diff
@@ -51,7 +52,56 @@ impl Storable for AccessRights {
     };
 }
 
-#[derive(CandidType, Deserialize, PartialEq, Eq, PartialOrd, Ord, Clone, Debug)]
+impl AccessControl for AccessRights {
+    fn can_read(&self) -> bool {
+        matches!(
+            self,
+            AccessRights::Read | AccessRights::ReadWrite | AccessRights::ReadWriteManage
+        )
+    }
+
+    fn can_write(&self) -> bool {
+        matches!(
+            self,
+            AccessRights::ReadWrite | AccessRights::ReadWriteManage
+        )
+    }
+
+    fn can_get_user_rights(&self) -> bool {
+        matches!(self, AccessRights::ReadWriteManage)
+    }
+
+    fn can_set_user_rights(&self) -> bool {
+        matches!(self, AccessRights::ReadWriteManage)
+    }
+
+    fn owner_rights() -> Self {
+        AccessRights::ReadWriteManage
+    }
+}
+
+pub trait AccessControl:
+    CandidType
+    + Serialize
+    + for<'a> Deserialize<'a>
+    + Clone
+    + Copy
+    + PartialEq
+    + Eq
+    + PartialOrd
+    + Ord
+    + std::fmt::Debug
+    + strum::IntoEnumIterator
+    + Storable
+{
+    fn can_read(&self) -> bool;
+    fn can_write(&self) -> bool;
+    fn can_get_user_rights(&self) -> bool;
+    fn can_set_user_rights(&self) -> bool;
+    fn owner_rights() -> Self;
```

Comment:
Is this `owner_rights` really needed? It seems it is used in a few `ensure_user_can*` methods, but the results of these methods seems never used: it is just used with `?`, i.e., it is just checked that the result is not an error but if the result is `Ok` then it is discarded.

Reply:
> We may want to find a less confusing construct to implement this though.

Yes, this is indeed somewhat confusing. This seems worth a ticket/to be improved. Ideally, the `ensure_*` methods simply return `Result<(), _>`.